libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -347,6 +373,24 @@ void AP_GPS::init(const AP_SerialManager& serial_manager)
     }
 }
 
+// set and save the device id parameter for the instance
+void AP_GPS::set_and_save_devid(int32_t instance, uint8_t value)
```

Comment:
Why do you need this helper? The point of the `friend class` stuff is to avoid one line wrappers like this, and the actual wrapper is implying the operation is much more complicated then it actually is.